### PR TITLE
Hacken and Certik Audit Fixes

### PR DESCRIPTION
**Pull request by narayanprusty**
## Description

This PR resolves the following audit fixes:

- VEN-958: VPB-06 | Missing Zero Address Validation
- VEN-965: VPB-15 | Unneccessary Use of Payable Casting
- VEN-1009: VPB-10 | `initialize()` Is Unprotected And Inconsistent Init Function
- VEN-1028: M07. Contradiction - NatSpec Contradiction
- VEN-1029: M08. Contradiction - Name Contradiction
- VEN-1031: M10. Inconsistent Data - Missing Event for Critical Value Updates
- VEN-1033: M12. Contradiction - NatSpec Comments Contradiction
- VEN-1034: L01. Floating Pragma
- VEN-1035: L02. Redundant Import
- VEN-1036: L03. Missing Zero Address Validation
- VEN-1037: L05. Contradiction

## Checklist

- [x] I have updated the documentation to account for the changes in the code.
- [x] If I added new functionality, I added tests covering it.
- [x] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [x] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/VToken.sol [hunk 1]
```diff
@@ -581,7 +581,7 @@ contract VToken is Ownable2StepUpgradeable, VTokenInterface, ExponentialNoError,
 
         /*
          *  We call `_doTransferIn` for the minter and the mintAmount.
-         *  Note: The vToken must handle variations between ERC-20 and ETH underlying.
+         *  Note: The vToken must handle variations of ERC-20 underlying.
```
Comment (chechu) [inline contracts/VToken.sol:584]: What would this sentence mean? I think we can remove the sentence

Comment (narayanprusty) [reply contracts/VToken.sol]: Agree. I have removed the sentence.

**Comment by narayanprusty** [comment]
> I don't see the changes in ProtocolShareReserve.sol and RewardsDistributor.sol associated to VEN-1009

I have pushed the changes now.

**Comment by UrAvgDeveloper** [comment]
LGTM !

**Review by UrAvgDeveloper — APPROVED** [review]

File: tests/hardhat/Rewards.ts [hunk 2]
```diff
@@ -283,20 +283,16 @@ describe("Rewards: Tests", async function () {
...
   it("Cannot add reward distributors with duplicate reward tokens", async function () {
     const RewardsDistributor = await ethers.getContractFactory("RewardsDistributor");
-    rewardsDistributor = await RewardsDistributor.deploy();
+    rewardsDistributor = await upgrades.deployProxy(RewardsDistributor, [comptrollerProxy.address, xvs.address]);
 
-    const rewardsDistributorDuplicate = await RewardsDistributor.deploy();
-    await rewardsDistributorDuplicate.initialize(comptrollerProxy.address, xvs.address);
-
-    await await expect(comptrollerProxy.addRewardsDistributor(rewardsDistributorDuplicate.address)).to.be.revertedWith(
+    await await expect(comptrollerProxy.addRewardsDistributor(rewardsDistributor.address)).to.be.revertedWith(
```
Comment (chechu) [inline tests/hardhat/Rewards.ts:288]: Duplicated `await`

Comment (narayanprusty) [reply tests/hardhat/Rewards.ts]: Removed

**Review by chechu — APPROVED** [review]

**Review by 0xlucian — APPROVED** [review]

**Pull request merged**